### PR TITLE
auto-generate hmn_connections and switch_metadata, also add shcd integration test

### PR DESCRIPTION
**Pull request by jacobsalmela**
#### Summary and Scope

- Fixes MTL-1483
- Relates to MTL-1466

##### Issue Type

- RFE Pull Request

Auto-generate seed files from canu-generated SHCDs.

#### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I tested this on internal system (x) (if yes, please include results or a description of the test)
 
#### Idempotency
 
N/a
 
#### Risks and Mitigations
 
Low

### CODE REVIEW

File: cmd/shcd.go [hunk 1]
```diff
@@ -157,6 +156,209 @@ type Switch struct {
 	Brand string
 }
 
+// Crafts and prints the xname of a give Id type in the SHCD
```
Comment (heemstra) [inline cmd/shcd.go:159]: nit: s/give/given/

Comment (jacobsalmela) [reply cmd/shcd.go:159]: Wilco

**Comment by jacobsalmela** [comment]
I'm going to pull in @alvarez3-hpe 's contribution in #75 since it ties in very closely.

**Review by rustydb — APPROVED** [review]

File: cmd/shcd_test.go [hunk 2]
```diff
@@ -18,123 +18,35 @@ import (
...
 )
 
-const _schemaFile = "../internal/files/shcd-schema.json"
+const _schema = "shcd-schema.json"
 
+var _schemaFile = filepath.Join("../internal/files", _schema)
+
+var switch_meta_expected = "../testdata/expected/" + switch_metadata
+var hmn_conn_expected = "../testdata/expected/" + hmn_connections
+var app_node_expected = "../testdata/expected/" + application_node_config
```
Comment (alvarez3-hpe) [inline cmd/shcd_test.go:27]: When do these testdata files get created?

Comment (jacobsalmela) [reply cmd/shcd_test.go:27]: They are part of this PR.  They care made from the valid JSON file

**Review by heemstra — APPROVED** [review]

**Pull request merged**